doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+[here](https://help.github.com/en/articles/fork-a-repo) (remember to sign
+into your account first). After this, you should see this copy in the list
+of your own repositories with a hint of it's origin.
+
+## Getting the source-code
+
+To develop for libelektra, we now have to "download" your copy of it's
+original repository. In git this process is called "cloning". Open a
+terminal and navigate to the folder you want to save the source code into
+and type:
+
+```sh
+git clone https://github.com/ElektraInitiative/libelektra.git
+```
+
+With the project now locally available we can start developing.
+
+## Setting up the project
+
+It is important to start CLion as root-user, since some functionalities of
+_kdb_ require superuser privileges to work as expected. Once you see the
+main menu, click "Open" and select the CMakeLists.txt-file inside the
+project's root directory. This will import the project accordingly and
+populate you run-configuration with some predefined values. In rare
+occurrences this won't happen. If that is the case for you, simply restart
+CLion using:
+
+_File_ --> _Invalidate Caches / Restart..._ --> _Invalidate and Restart_
+
+Now after all processes of CLion have finished, the project should be set up
+and the run-configuration should be populated with entries.
+
+## Development
+
+Usually the folders you have to work in to add functionality or
+documentation are as follows:
+
+- doc<br/>
+  This folder contains mainly all documentation of the project, including
+  almost all pages of the [homepage](https://www.libelektra.org) of this
+  project. One important note is, that all markdown-pages can also be used
+  for testing using
+  [Markdown to Shell Recorder](https://github.com/ElektraInitiative/libelektra/tree/master/tests/shell/shell_recorder/tutorial_wrapper)
+  (you can find an example on how to do this [here](/doc/help/kdb-get.md)).
+- src<br/>
+  Almost all functionality-code resides here.
+  - bindings<br/>
+    Here is all the code of available [Bindings](/src/bindings/README.md)
+    of libelektra. - plugins<br/>
+    You can find all developed [Plugins](/src/plugins/README.md) here. If
+    you want to fix a bug for an existing plugin or add another one, this is
+    the directory you have to work in. For further information on how to
+    develop your own plugin, please visit [this](/doc/tutorials/plugins.md)
+    tutorial. - tools<br/>
+    In this folder the source of all tools for interacting with Electra's
+    global key database, e.g. _kdb_, can be found.
+- tests<br/>
+  Here you can find all sorts of tests (excluding tests created with the
+  _Markdown to Shell Recorder_-tool).
+
+## Testing
+
+The most thorough way to test your changes is to run all tests. Therefore
+navigate to your run-configurations (Run -> Edit Configurations...) and look
+for the entry "run*all". There, "run_all" should be selected as \_Executable*,
+"kdb" as _Target_. Now you can execute this run-configuration which will run
```

Comment:
Actually for `run_all` you should probably select `All targets` (right at the top of the list) to ensure everything is built. Although I never tried running `run_all` this way. I simply use the integrated terminal and run `make run_all` in the cmake-build-xyz folder.